package/MDAnalysis/analysis/align.py

```diff
@@ -380,6 +377,175 @@ def alignto(mobile, reference, select="all", mass_weighted=False,
+             and setting different parameters as described for the corresponding writer).
+
+
+            .. _ClustalW: http://www.clustal.org/
+            .. _STAMP: http://www.compbio.dundee.ac.uk/manuals/stamp.4.2/
+
+            .. versionchanged:: 0.8
+               Added *kwargs* to be passed to the trajectory :class:`~MDAnalysis.coordinates.base.Writer` and
+               *filename* is returned.
+
+            .. versionchanged:: 0.10.0
+               Uses :func:`get_matching_atoms` to work with incomplete selections
+               and new *strict* keyword. The new default is to be lenient whereas
+               the old behavior was the equivalent of *strict* = ``True``."""
+
+        frames = u.trajectory
+        self._setup_frames(frames, start,stop,step)
+        self.rmsdfile = rmsdfile
+
+        if quiet:
+            # should be part of a try ... finally to guarantee restoring the log level
+            logging.disable(logging.WARN)
+
+        kwargs.setdefault('remarks', 'RMS fitted trajectory to reference')
+
+        if filename is None:
+            path, fn = os.path.split(frames.filename)
+            self.filename = os.path.join(path, prefix + fn)
+            _Writer = frames.Writer
+        else:
+            _Writer = frames.OtherWriter
+        if os.path.exists(self.filename) and not force:
+            logger.warn("{0} already exists and will NOT be overwritten; use force=True if you want this".format(self.filename))
+
+        self.writer = _Writer(self.filename, **kwargs)
+        del _Writer
+        select = rms._process_selection(select)
+        ref_atoms = reference.select_atoms(*select['reference'])
+        traj_atoms = u.select_atoms(*select['mobile'])
+        natoms = traj_atoms.n_atoms
+
+        ref_atoms, traj_atoms = get_matching_atoms(ref_atoms, traj_atoms,
+                                        tol_mass=tol_mass, strict=strict)
+
+        logger.info("RMS-fitting on {0:d} atoms.".format(len(self.ref_atoms)))
+
+        if mass_weighted:
+            # if performing a mass-weighted alignment/rmsd calculation
+            weights = ref_atoms.masses
+        else:
+            weights = None
+
+        # reference centre of mass system
+        ref_com = ref_atoms.center_of_mass()
+        ref_coordinates = ref_atoms.positions - nref_com
+
+        # allocate the array for selection atom coords
+        traj_coordinates = traj_atoms.positions.copy()
+
+        nframes = len(frames)
+        self.rmsd = np.zeros((1,))
+
+
+        # R: rotation matrix that aligns r-r_com, x~-x~com
+        #    (x~: selected coordinates, x: all coordinates)
+        # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
+        rot = np.zeros(9, dtype=np.float64)  # allocate space for calculation
+        R = np.matrix(rot.reshape(3, 3))
+
+        self.percentage = ProgressMeter(nframes, interval=10, quiet=quiet,
```

Comment:
I'd much rather add in `ProgressMeter` to `AnalysisBase` than have every class define their own